File: web/svelte/src/lib/stores/modal-store.ts
import { writable } from 'svelte/store';

export enum ModalMode {
  edit = 'edit',
  add = 'add'
}

interface ModalStore {
  open: boolean;
  mode: ModalMode;
  name: string | undefined;
  url: string | undefined;
}

const modalStoreInstance = writable<ModalStore>({
  open: false,
  mode: ModalMode.add,
  name: undefined,
  url: undefined
});

export const modalStore = {
  subscribe: modalStoreInstance.subscribe,
  open: ({
    mode = ModalMode.add,
    name,
    url
  }: { mode?: ModalMode; name?: string; url?: string } = {}) =>
    modalStoreInstance.set({
      open: true,
      mode,
      name,
      url
    }),
  close: () =>
    modalStoreInstance.set({
      open: false,
      mode: ModalMode.add,
      name: undefined,
      url: undefined
    }),
  set: modalStoreInstance.set
};

File: apple/TRssReader/Stores/ModalStore.swift
//
//  ModalStore.swift
//  TRssReader
//
//  Created by Ty Hopp on 21/4/23.
//

import Foundation

@MainActor
final class ModalStore: ObservableObject {
    enum Mode {
        case add
        case edit
    }
    
    @Published var isOpen: Bool = false
    @Published var mode: Mode = .add
    @Published var name: String = ""
    @Published var url: String = ""
    
    func open(mode: Mode = .add, name: String = "", url: String = "") {
        self.isOpen = true
        self.mode = mode
        self.name = name
        self.url = url
    }
    
    func close() {
        self.isOpen = false
        self.mode = .add
        self.name = ""
        self.url = ""
    }
}
